tests/ssl-opt.sh

```diff
@@ -543,16 +543,18 @@ print_name() {
 
 # record_outcome <outcome> [<failure-reason>]
 # The test name must be in $NAME.
+# Use $TEST_SUITE_NAME as the test suite name if set.
 record_outcome() {
     echo "$1"
     if [ -n "$MBEDTLS_TEST_OUTCOME_FILE" ]; then
         printf '%s;%s;%s;%s;%s;%s\n' \
                "$MBEDTLS_TEST_PLATFORM" "$MBEDTLS_TEST_CONFIGURATION" \
-               "ssl-opt" "$NAME" \
+               "${TEST_SUITE_NAME:-ssl-opt}" "$NAME" \
                "$1" "${2-}" \
                >>"$MBEDTLS_TEST_OUTCOME_FILE"
     fi
 }
+unset TEST_SUITE_NAME
```

Comment:
I am not sure why we would need this here. There is already such an unset line 9024.

Reply:
In case it's set in the environment when the script starts. Line 9024 unsets it after it's been set internally.

Reply:
ok got it, wouldn't there be a better place for it ? Beginning of the file ? It is a bit in the middle of nowhere here.

Reply:
I put it next to the place where it's used for the first time.

Reply:
ok personally I would have put it rather at the beginning of the file where other environment variables are initialized but I do not have a strong opinion about it thus approving.